checkstyle-tester/projects-to-test-on-for-github-action.properties

```diff
@@ -40,5 +40,4 @@ nbia-dcm4che-tools|git|https://github.com/thprakash/nbia-dcm4che-tools|c3591e6f0
 # RequireThis usage
 spring-integration|git|https://github.com/spring-projects/spring-integration|main|
 
-# openjdk 17 requires lots of excludes, list here should be consistent with file filters at https://github.com/checkstyle/checkstyle/blob/master/.ci/openjdk17-excluded.files
```

Comment:
@nrmancuso This line says it must have excludes consistent with main repo. Are you sure we can remove all excludes?

Reply:
Related discussions:

https://github.com/checkstyle/checkstyle/pull/10956

https://github.com/checkstyle/contribution/pull/594

Reply:
I tried to catch up with discussions

We used to have an exclude list (non-compilable file) for all projects in this file. [but we then removed all excludes from all projects](https://github.com/checkstyle/contribution/pull/584#issuecomment-986359760) because it turned out that these excluded files were ignored. We removed them to be consistent with previous reports. 

After that, [we added support for exclude option with `--allowExcludes`](https://github.com/checkstyle/checkstyle/pull/10956). But we never returned the exclude list to the projects list. 

Then [we added this openjdk17 with this huge exclude list](https://github.com/checkstyle/contribution/pull/594). 

I can confirm that this file `(projects-to-test-on-for-github-action.properties)` isn't used by any other task like no-exception. It is only used as a `DEFAULT_PROJECTS_LINK` in diff-report.yml. So, removing this exclude list and openjdk17 will not affect any other task.

now the question should we add an exclude list for the new openjdk21? If yes, then why don't we add an exclude list for the other projects?

> Should we add an exclude list for the new openjdk21?

I don't think we gain any thing if we add it. we have 2 cases :

- non-compilable file and can't be parsed by us: This will get ignored due to the `skipFileOnJavaParseException` new treewalker property 

- non-compilable file but can be parsed ([example](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/f5590ec_2022232703/reports/diff/openjdk17/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/openjdk17/test/langtools/tools/javac/diags/examples/DuplicateTotalPattern.java.html#L31)): This will have no parse error so will not be skipped and it will appear in the report. Is this acceptable?  for me, I don't think we will lose anything by not excluding those files On the contrary they might help ensure regression ([example for a  non-compilable file that shows that the check update is as expected](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/f5590ec_2022232703/reports/diff/openjdk17/index.html#A1)  from [this PR](https://github.com/checkstyle/checkstyle/pull/11221#issue-1108696113)) 
 > I am not sure if we should add files filters for these files, though, since they are how I confirmed logic of check update.

Reply:
@nrmancuso 
Please correct me if anything is wrong I tried to go quickly over the blame of this file. You must have strong opinion because all these PR are created by you :)

Reply:
>non-compilable file but can be parsed ([example](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/f5590ec_2022232703/reports/diff/openjdk17/xref/home/runner/work/checkstyle/checkstyle/contribution/checkstyle-tester/repositories/openjdk17/test/langtools/tools/javac/diags/examples/DuplicateTotalPattern.java.html#L31)): This will have no parse exception so will not be skipped and it will appear in the report. Is this acceptable? for me, I don't think we will lose anything by not excluding those

Thanks a lot for such detailed post.
There is a nuance. If file is not compile able by javac, but parse able by checkstyle - target Check will execute and produce some violations or no violations.
It will be very confusing, as in report it will looks like diff in behavior and will be red flag for all reviewers. It will be not easy to catch a fact that this file is not legitimate to run on.

Removal of excludes is prone to cause bunch of false investihations for author of PR or bad requests from maintainers to investigate differences.

I think we need to keep exclude list for open jdk.

Reply:
I forgot to ask a question in my post.

> file is not compile able by javac, but parse able by checkstyle

How do we even know these files?

Reply:
Okay looks like this huge exclude list for non-parseable files by CS only. It didn't include the first case (file is not compile able by javac, but parse able by checkstyle) I discovered this in a very recent [report](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/b2743e3_2024192510/reports/diff/openjdk17/index.html#A563) it include a non compile able but parse able file. I think it makes sense we have no way to discover such files. In this case, the excludes is for non-parseable files only which can be completely ignored by the treewalker property 

@romani please reconsider,  can we find any way to know not compilable by javac, but parse able by checkstyle files

Reply:
> How do we even know these files?

it was hard work of @nrmancuso .

>  can we find any way to know not compilable by javac, but parse able by checkstyle files

only by pratice and spotting something weird in diff report.
We can make a script to find all, but I am not sure that it is reasonable to do this at summer.

> In this case, the excludes is for non-parseable files only which can be completely ignored by the treewalker property

should be skipped completely for diff report.
Treewalker can handle all, this list is for known files that can give weird behavior and confuse readers of diff report.

Reply:
@nrmancuso , please step in.

Reply:
These exclusions are to avoid having to analyze files that don't matter (noncompilable) or that we don't support parsing. It is generated by bash/python from [here](https://github.com/checkstyle/checkstyle/blob/f5c492af3fde6bd417053e2f5a0d187767b3551b/config/projects-to-test/openjdk21-excluded.files#L1).

Let's just generate these and move on, we can debate the value of having these in another issue

Reply:
Look like @nrmancuso , agree to keep this list but content might be different.

Reply:
@nrmancuso @romani done please approve
issue : https://github.com/checkstyle/contribution/issues/881